analyzers/tests/SonarAnalyzer.UnitTest/TestCases/InvocationResolvesToOverrideWithParams.cs

```diff
@@ -2,161 +2,408 @@
+        string.Concat("aaaa"); // Noncompliant, resolves to params, but there's a single object version too.
 
-            Console.WriteLine("format", 0, 1, "", ""); //Compliant
-        }
+        Console.WriteLine("format", 0, 1, "", ""); // Compliant
     }
+}
+
+public class MyClass
+{
+    public void Format(string a, params object[] b) { }
+    public void Format() { } // The presence of this method causes the issue
 
-    public class MyClass
+    public void Test()
     {
-        public void Format(string a, params object[] b) { }
-        public void Format() { } // The presence of this method causes the issue
+        Format("", null, null);
+    }
+}
 
-        public void Test()
-        {
-            Format("", null, null);
-        }
+public class Test
+{
+    public void MyMethod(params string[] s) { }
+    public void MyMethod(Test s) { }
+
+    public Test()
+    {
+        MyMethod(""); // Compliant
     }
+}
+
+public class Test2
+{
+    public static implicit operator Test2(string s) { return null; }
+
+    public void MyMethod(params string[] s) { }
+    public void MyMethod(Test2 s) { }
 
-    public class Test
+    public Test2()
     {
-        public void MyMethod(params string[] s) { }
-        public void MyMethod(Test s) { }
+        MyMethod(""); // Noncompliant
+    }
+}
 
-        public Test()
-        {
-            MyMethod(""); // Compliant
-        }
+// See https://github.com/SonarSource/sonar-dotnet/issues/2234
+public class FuncAndActionCases
+{
+    static void Main(string[] args)
+    {
+        M1(() => Console.WriteLine("hi"));
     }
 
-    public class Test2
+    public static void M1(params Action[] a) { }
+    public static void M1<T>(Func<T> f) { }
+    public static void M1(Func<Task> f) { }
+}
+
+public class WithLocalFunctions
+{
+    public static void Test(int foo, params object[] p)
     {
-        public static implicit operator Test2(string s) { return null; }
+        Console.WriteLine("test1");
+    }
 
-        public void MyMethod(params string[] s) { }
-        public void MyMethod(Test2 s) { }
+    public static void Test(double foo, object p1)
+    {
+        Console.WriteLine("test2");
+    }
 
-        public Test2()
+    public void Method()
+    {
+        static void Call()
         {
-            MyMethod(""); // Noncompliant
+            Test(42, null); // Noncompliant {{Review this call, which partially matches an overload without 'params'. The partial match is 'void WithLocalFunctions.Test(double foo, object p1)'.}}
         }
     }
+}
 
-    // See https://github.com/SonarSource/sonar-dotnet/issues/2234
-    public class FuncAndActionCases
+// Reproducer for https://github.com/SonarSource/sonar-dotnet/issues/5430
+namespace Repro5430
+{
+    public class SomeClass
     {
-        static void Main(string[] args)
+        private int InlineInitializedField11 = new SomeClass("1", "s1").PrivateOverload("s1");                  // Compliant
```

Comment:
This would have been enough, the rest is obvious
```suggestion
        private int Field11 = new SomeClass("1", "s1").PrivateOverload("s1");                  // Compliant
```

Reply:
Also below